batch/batch/front_end/templates/job.html

```diff
@@ -50,6 +50,22 @@ <h2>Attempts</h2>
 {% endif %}
 
 <h2>Step Status</h2>
+<section class="progress_bar">
+  {% for step in step_statuses %}
+{% if step %}
+  {% for name, timing in step['timing'].items() %}
+  <div class="main">
+  <div class="bar" style="width:{{timing['duration']/50}}px">
```

Comment:
I noticed when a job is in-progress, this line causes a 500 error:
```
{"asctime":"2021-02-26 01:16:26,959","container":"batch-68d5b87f4b-6btjn","x_real_ip":null,"connection_id":null,"exc_info":"Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/aiohttp/web_protocol.py", line 418, in start
    resp = await task
  File "/usr/local/lib/python3.7/dist-packages/aiohttp/web_app.py", line 458, in _handle
    resp = await handler(request)
  File "/usr/local/lib/python3.7/dist-packages/aiohttp/web_middlewares.py", line 119, in impl
    return await handler(request)
  File "/usr/local/lib/python3.7/dist-packages/aiohttp/web_middlewares.py", line 119, in impl
    return await handler(request)
  File "/usr/local/lib/python3.7/dist-packages/aiohttp_session/__init__.py", line 152, in factory
    response = await handler(request)
  File "/usr/local/lib/python3.7/dist-packages/prometheus_async/aio/_decorators.py", line 42, in time_decorator
    rv = await wrapped(*args, **kw)
  File "/usr/local/lib/python3.7/dist-packages/gear/auth.py", line 92, in wrapped
    return await fun(request, userdata, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/batch/front_end/front_end.py", line 154, in wrapped
    return await fun(request, userdata, batch_id, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/batch/front_end/front_end.py", line 1334, in ui_get_job
    return await render_template('batch', request, userdata, 'job.html', page_context)
  File "/usr/local/lib/python3.7/dist-packages/web_common/web_common.py", line 88, in render_template
    response = aiohttp_jinja2.render_template(file, request, context)
  File "/usr/local/lib/python3.7/dist-packages/aiohttp_jinja2/__init__.py", line 73, in render_template
    text = render_string(template_name, request, context, app_key=app_key)
  File "/usr/local/lib/python3.7/dist-packages/aiohttp_jinja2/__init__.py", line 64, in render_string
    text = template.render(context)
  File "/usr/local/lib/python3.7/dist-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/local/lib/python3.7/dist-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.7/dist-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.7/dist-packages/batch/front_end/templates/job.html", line 1, in top-level template code
    {% extends "layout.html" %}
  File "/usr/local/lib/python3.7/dist-packages/web_common/templates/layout.html", line 25, in top-level template code
    {%block content %}{% endblock %}
  File "/usr/local/lib/python3.7/dist-packages/batch/front_end/templates/job.html", line 58, in block "content"
    <div class="bar" style="width:{{timing['duration']/50}}px">
TypeError: unsupported operand type(s) for /: 'NoneType' and 'int'","message":"Error handling request"}
```

The step_stauses data used by `job.html` is the same data returned by the jobs API endpoint. You can use `hailctl curl` to directly look at this information. Below I run `hailctl curl ...` to request the data from the `batch` in my namespace (`dking`) for job `1` in batch `16781`.

I piped the output of curl into [`jq`](https://stedolan.github.io/jq/) and used jq to extract just the container statuses. The container statuses are the same as the `step_statuses`.

Notice that the job is still `"running"`. When this happens, the `"running"` entry of the `"timing"` dictionary *doesn't have a duration*. That's why I get the error.
```json
(base) # hailctl curl dking batch '/api/v1alpha/batches/16781/jobs/1' | jq '.status.container_statuses'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2070  100  2070    0     0   5048      0 --:--:-- --:--:-- --:--:--  5036
{
  "main": {
    "name": "main",
    "state": "running",
    "timing": {
      "pulling": {
        "start_time": 1614301891698,
        "finish_time": 1614301894555,
        "duration": 2857
      },
      "creating": {
        "start_time": 1614301894555,
        "finish_time": 1614301894814,
        "duration": 259
      },
      "starting": {
        "start_time": 1614301894825,
        "finish_time": 1614301895305,
        "duration": 480
      },
      "running": {
        "start_time": 1614301895305
      }
    },
    "container_status": {
      "state": "running",
      "started_at": "2021-02-26T01:11:35.285970352Z",
      "finished_at": "0001-01-01T00:00:00Z",
      "out_of_memory": false,
      "exit_code": 0
    }
  }
}
```

I think a simple fix is this:
```
  {% if 'duration' in timing %}
  <div class="main">
  <div class="bar" style="width:{{timing['duration']/50}}px">
  ...
  {% endif %}
```